From a9aaff5b5b94e76431126036ee0cc41a3bab8218 Mon Sep 17 00:00:00 2001
From: Venkatasaiprudhvi Raj Sannidhi <vsannidhi@maxlinear.com>
Date: Fri, 5 Sep 2025 00:13:32 -0700
Subject: [PATCH] WLANRTSYS-93880 [WAV700][MBSSID] PN number for non tx VAP is
 getting fetched incorrectly

Description: Remove the code which is not necessary.

Unit test: Pass
---
 src/ap/wpa_auth.c | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/src/ap/wpa_auth.c b/src/ap/wpa_auth.c
index cef46fa00..6f029e23a 100644
--- a/src/ap/wpa_auth.c
+++ b/src/ap/wpa_auth.c
@@ -209,16 +209,11 @@ static inline int wpa_auth_get_seqnum(struct wpa_authenticator *wpa_auth,
 {
 	int res;
 #ifdef CONFIG_VENDOR_MXL
-	struct wpa_authenticator *wpa_auth_backup = wpa_auth;
-
 	if (wpa_auth->conf.mxl_tx_bss_auth != NULL)
 		wpa_auth = wpa_auth->conf.mxl_tx_bss_auth;
 #endif /* CONFIG_VENDOR_MXL */
 
 	if (!wpa_auth->cb->get_seqnum)
-#ifdef CONFIG_VENDOR_MXL
-		wpa_auth = wpa_auth_backup;
-#endif /* CONFIG_VENDOR_MXL */
 		return -1;
 	res = wpa_auth->cb->get_seqnum(wpa_auth->cb_ctx, addr, idx, seq);
 #ifdef CONFIG_TESTING_OPTIONS
@@ -242,10 +237,6 @@ static inline int wpa_auth_get_seqnum(struct wpa_authenticator *wpa_auth,
 			  WPA_KEY_RSC_LEN);
 	}
 #endif /* CONFIG_TESTING_OPTIONS */
-
-#ifdef CONFIG_VENDOR_MXL
-	wpa_auth = wpa_auth_backup;
-#endif /* CONFIG_VENDOR_MXL */
 	return res;
 }
 
-- 
2.43.0

